pal/src/map/virtual.cpp

```diff
@@ -2002,6 +2025,7 @@ VirtualProtect(
     MemSize = (((UINT_PTR)(dwSize) + ((UINT_PTR)(lpAddress) & VIRTUAL_PAGE_MASK)
                 + VIRTUAL_PAGE_MASK) & ~VIRTUAL_PAGE_MASK);
 
+#if DEBUG
```

Comment:
Curious: Why is this under DEBUG? Is the claim that we'll always call VirtualProtect with valid flags (and fail on debug if we don't?) Is it too costly to do a runtime check here? VirtualProtect calls should be rare (not on the hot path), and given the rest of the cost of VirtualProtect, having this runtime protection seems worthwhile? (Because otherwise, the behavior with invalid flags is more dangerous, and could cause security issues if an attacker can somehow corrupt the flags we pass into VirtualProtect)

Reply:
No, not because it is too costly. Because it's kinda redundant. Debug check should be just fine. no?

Reply:
Curious- where is the cost here? I assume other things this function does (like call mprotect and VIRTUALFindRegionInformation) would be way more expensive than checking if the flags are valid, no?

Reply:
You are right. `mprotect` call is incomparably more expensive. As I mentioned, took them off because of they are rather redundant and not sure how cool we are with them on prod. mode.